Makefile

```diff
@@ -14,6 +14,7 @@ CUDA_OUTPUT_FILE = -o $@
 # NVCC flags
 # -t=0 is short for --threads, 0 = number of CPUs on the machine
 NVCC_FLAGS = -O3 -t=0 --use_fast_math
+NVCC_FLAGS += --std=c++17
```

Comment:
I'm not 100% sure what the repercussions would be here right now, is it needed for the modal script?

Reply:
Currently yes, it prints out 100s of lines of errors in the terminal without that (I believe its because of the docker image I created having the newest version of cuda), but the current script will not work without that.

<img width="1293" alt="Screenshot 2024-06-04 at 9 30 24 AM" src="https://github.com/karpathy/llm.c/assets/93402393/e4e70e2b-3360-477c-b3df-8311051d45f8">

Reply:
What version of GCC are you using?

Reply:
As in on my container? I am unsure because I used a cuda image from docker hub as a base, but the current image I am using is this: 

[https://hub.docker.com/layers/totallyvyom/cuda-env/latest/images/sha256-a273f155a5ad9dcd4dd366a6da2f48f80f10b675bd136ab350f73b695d23b23b?tab=layers](url)

and this is the cuda image I am layering on:

[https://hub.docker.com/layers/nvidia/cuda/12.4.1-cudnn-devel-ubuntu20.04/images/sha256-f18cf1a9ac2842e59f13b0d0729594da8cbd68cadd2379308cdd98c0374dbd80?context=explore](url)

Reply:
That helps - it's probably (I'm guessing) the ubuntu 20.04 compiler is too old. Try finding a container which has a newer version of ubuntu like 22.04. That should fix your compilation issues.

Reply:
I am messing around with images for newer versions to also fix the errors I was getting with nsight, but it takes while because reloading all the containers after every little detail takes so much time, it is such a hassle to build a cuda dev env on a docker 😅

Reply:
Yep, your right on that. Good luck!